source/slang/core.meta.slang

```diff
@@ -870,6 +870,14 @@ bool operator!=(__none_t noneVal, Optional<T> val)
     return val.hasValue;
 }
 
+__generic<each T>
+__magic_type(TupleType)
+struct Tuple
+{
+    __intrinsic_op($(0))
+    __init(expand each T);
+}
```

Comment:
This is a question.
This implementation is a kind of opaque type whose implementation is not visible to the end users.
Isn't the variadic argument feature good enough to make this not opaque?
I asked the same question on the other review but I am still not sure if there is a reason why the implementation is less on frontend side and more on the c++ side.
```
struct Tuple<T0,T> : Tuple<T>
{
 T0 v;
 ...
}
```
My guess is that currently the implementation for the swizzle and constructor needs to be implemented on c++ side due to a certain limitations Slang still have.
Does this mean the users cannot declare their own version of custom Tuple-like struct because we are still missing certain language feature to support cases like Tuple?

Reply:
No matter what we do, there needs to have a built in mechanism to represent storage for a type pack, and tuple types is that built in mechanism so we have no way to define it in user space. If the user wish to define a variable that holds values for a list of types, they will use tuple types for that. Tuple type itself is the language mechanism to define storage for type packs so it has to be implemented as builtin functionality. 

And you are right that the user cannot define another thing like tuple type where fields are named differently.